### PR TITLE
Bidirectional infinite scroll

### PR DESCRIPTION
**Pull request by jonathanjma**
- Refactored Scrollable calendar to support bidirectional infinite scrolling
- If URL contains the ```date``` query parameter, the calendar will start at the provided date.
- Scrolling up will now result in more recent entries being displayed (until the current date is reached)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "preline": "^1.9.0",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
-    "react-infinite-scroll-component": "^6.1.0",
+    "react-intersection-observer": "^9.5.3",
```
Comment (zachseidner1) [inline package.json:16]: nice job finding this library and applying it this way!

**Review by zachseidner1 — APPROVED** [review]
Amazing work Jonathan! Seriously, you were able to implement a challenging task quickly and efficiently with minimal code. All the code looks good and the interaction is clean. PR approved 💰

could you add the demo video you sent to the PR description? Just to improve documentation a little.

File: package.json (hunk 1 above)
Comment (jonathanjma) [reply package.json:16]: Found it in an react query example lol

File: src/utils.ts [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
   return `${year}-${month}-${day}`;
 }
 
+// Strips time from Date and shifts the day by the given amount
+export const modifyDateDay = (date: Date, dayDiff: number) =>
+  new Date(date.getFullYear(), date.getMonth(), date.getDate() + dayDiff);
+
+// Convert YYYY-MM-DD date string to Date and strip time
+export const dateFromStr = (dateStr: string) => new Date(dateStr + "T00:00:00");
+
```
Comment (zachseidner1) [inline src/utils.ts:17]: Change these to `export function` for codebase consistency! or change everything else to const if you really want to lol. Consistency is what matter most.

**Review by zachseidner1 — CHANGES_REQUESTED** [review]

**Review by zachseidner1 — APPROVED** [review]
thanks

**Pull request merged**